src/generic/grid.cpp

```diff
@@ -2851,10 +2851,14 @@ void wxGrid::Init()
     m_gridLinesEnabled = true;
     m_gridLinesClipHorz =
     m_gridLinesClipVert = true;
-    m_cellHighlightColour = *wxBLACK;
+    m_cellHighlightColour = wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHTTEXT);
```

Comment:
I'm not very comfortable with this commit (which I'd like to discuss separately, as it's completely unrelated to the rest of the PR) because while it undoubtedly improves things under macOS, it risks breaking them under the other platforms, resulting in another round of changes to fix them later, breaking them for macOS, fixing them there, rebreaking them elsewhere etc...

In particular, this one is _white_ under MSW by default, so I'm pretty sure using it instead of the previously used black is not going to work well, i.e. it will result in invisible highlight.

I think it should be `wxSYS_COLOUR_WINDOWTEXT` instead, or maybe `wxSYS_COLOUR_LISTBOXTEST`...